akka-http-core/src/test/scala/akka/http/impl/model/parser/HttpHeaderSpec.scala

```diff
@@ -196,6 +196,28 @@ class HttpHeaderSpec extends AnyFreeSpec with Matchers {
         `Content-Disposition`(ContentDispositionTypes.attachment, Map("filename" -> "file/txt", "name" -> "field1"))
       "Content-Disposition: attachment; name=\"field1\"; other=\"\"" =!=
         `Content-Disposition`(ContentDispositionTypes.attachment, Map("name" -> "field1", "other" -> ""))
+      "Content-Disposition: attachment; filename=\"x=2*2+1(?)\"; filename*=UTF-8''x%3D2%2A2+1%28%E2%82%AC%29" =!=
```

Comment:
Is there a rule in which order `filename` and `filename*` should appear in the parameter list? Because currently the "accidental" solution seems to be that the last `filename` parameter "wins" (probably due to the `TreeMap` constructor in https://github.com/akka/akka-http/pull/3705/files#diff-ad128a726c2bd0651c962a329a9b0d62d2e763a6d76fc3fa5b96cf7084f3173bR23).

If there's no strict rule, could you also add tests going the other way round? I suspect they might be failing ("filename" value overwrites "filename*") in which case we need a smarter resolution rule in the parser.

Reply:
I think `TreeMap` was used to handle rendering, but I do not know the reasoning behind "last parameter wins" in parser. This should be fixed, too.

Reply:
Thanks